tests/em/tdem/test_TDEM_DerivAdjoint_RawWaveform.py

```diff
@@ -16,298 +17,360 @@
-                return [prb.survey.dpred(m), lambda mx: prb.Jvec(m, mx)]
-            print('\ntest_Jvec_{}_{}'.format(prbtype, rxcomp))
-            Tests.checkDerivative(derChk, m, plotIt=False, num=2, dx=m*2,
-                                  eps=1e-20)
+        def test_Jvec_h_dhdtx(self):
+            self.JvecTest('dhdtx')
 
-        def test_Jvec_b_bx(self):
-            self.JvecTest(prbtype='b', rxcomp='bx')
+        def test_Jvec_h_dhdtz(self):
+            self.JvecTest('dhdtz')
 
-        def test_Jvec_b_bz(self):
-            self.JvecTest(prbtype='b', rxcomp='bz')
+    if testAdjoint:
+        def test_Jvec_adjoint_h_hx(self):
+            self.JvecVsJtvecTest('hx')
 
-        def test_Jvec_b_dbxdt(self):
-            self.JvecTest(prbtype='b', rxcomp='dbdtx')
+        def test_Jvec_adjoint_h_hz(self):
+            self.JvecVsJtvecTest('hz')
 
-        def test_Jvec_b_dbzdt(self):
-            self.JvecTest(prbtype='b', rxcomp='dbdtx')
+        def test_Jvec_adjoint_h_dhdtx(self):
+            self.JvecVsJtvecTest('dhdtx')
 
-        def test_Jvec_b_ey(self):
-            self.JvecTest(prbtype='b', rxcomp='ey')
+        def test_Jvec_adjoint_h_dhdtz(self):
+            self.JvecVsJtvecTest('dhdtz')
 
-        def test_Jvec_e_ey(self):
-            self.JvecTest(prbtype='e', rxcomp='ey')
+        def test_Jvec_adjoint_h_jy(self):
+            self.JvecVsJtvecTest('jy')
 
-        def test_Jvec_e_dbdtx(self):
-            self.JvecTest(prbtype='e', rxcomp='dbdtx')
 
-        def test_Jvec_e_dbdtz(self):
-            self.JvecTest(prbtype='e', rxcomp='dbdtz')
+class DerivAdjoint_J(Base_DerivAdjoint_Test):
 
-        def test_Jvec_h_hx(self):
-            self.JvecTest(prbtype='h', rxcomp='hx')
+    formulation = 'j'
 
-        def test_Jvec_h_hz(self):
-            self.JvecTest(prbtype='h', rxcomp='hz')
+    if testDeriv:
+        def test_Jvec_j_jy(self):
+            self.JvecTest('jy')
 
-        def test_Jvec_h_dhdtx(self):
-            self.JvecTest(prbtype='h', rxcomp='dhdtx')
+        def test_Jvec_j_dhdtx(self):
+            self.JvecTest('dhdtx')
 
-        def test_Jvec_h_dhdtz(self):
-            self.JvecTest(prbtype='h', rxcomp='dhdtz')
+        def test_Jvec_j_dhdtz(self):
+            self.JvecTest('dhdtz')
 
-        def test_Jvec_h_jy(self):
-            self.JvecTest(prbtype='h', rxcomp='jy')
+    if testAdjoint:
+        def test_Jvec_adjoint_j_jy(self):
+            self.JvecVsJtvecTest('jy')
 
-        def test_Jvec_j_jy(self):
-            self.JvecTest(prbtype='j', rxcomp='jy')
+        def test_Jvec_adjoint_j_dhdtx(self):
+            self.JvecVsJtvecTest('dhdtx')
 
-        def test_Jvec_j_dhdtx(self):
-            self.JvecTest(prbtype='j', rxcomp='dhdtx')
+        def test_Jvec_adjoint_j_dhdtz(self):
+            self.JvecVsJtvecTest('dhdtz')
 
-        def test_Jvec_j_dhdtz(self):
-            self.JvecTest(prbtype='j', rxcomp='dhdtz')
 
+# class TDEM_DerivTests(unittest.TestCase):
```

Comment:
Would be nice to have these tests perhaps tested on dev-->master? Just so that we know that it is still working, because they are useful for debugging.